profiles/base.config.js

```diff
@@ -4,5 +4,6 @@ export default {
   moduleName: 'esriLoader',
   exports: 'named',
   dest: 'dist/umd/esri-loader.js',
-  sourceMap: true
+  sourceMap: true,
+  context: 'window'
```

Comment:
I do see that [we use this option in rest-js](https://github.com/Esri/arcgis-rest-js/blob/1a527a38a3596d4b9ad1a3d2122dd7bf5db1d3e8/umd-base-profile.js#L73), but I don't think we need this, do we? We're not using `this` w/in the modules, right? Do we need this for `let defaultOptions` in script.ts? I would not expect so. I would expect that would just end up as a `var` scoped to the IIFE that that module gets transpiled to.

Reply:
I added `content: 'window'` since `{ ...defaultOptions, ...options }` compiles down to the following:

```js
var __assign = (this && this.__assign) || function () {
                   ^
   __assign = Object.assign || function(t) {
       for (var s, i = 1, n = arguments.length; i < n; i++) {
           // ...
       }
   }
}
```

Which was throwing this warning:

>⚠️   The 'this' keyword is equivalent to 'undefined' at the top level

Reply:
Also, as a side note... I tried `Object.assign(defaultOptions, options)`, but that threw an error as well:

>Property 'assign' does not exist on type 'ObjectConstructor'.

Pretty sure that is due to the `tsconfig.json` target being set to `es5`, which is why I went with the spread option: `{ ...defaultOptions, ...options }`

Reply:
ah, I see. I'm surprised this hasn't come up yet

Let's replace this w/ a for...in w/ `hasOwnProperty` for now and add a comment above it like:

```ts
// we would have liked to use spread like { ...defaultOptions, ...options }
// but TS would inject a polyfill that would require use to configure rollup w content: 'window'
// if we have another occasion to use spread, let'd do that and replace this for...in
```

Reply:
Done 👍